contracts/base/BlockRewardHbbftBase.sol

```diff
@@ -201,8 +211,10 @@ contract BlockRewardHbbftBase is Initializable, OwnableUpgradeable, IBlockReward
         );
         // If this is the last block of the epoch i.e. master key has been generated.
 
-        if (_isEpochEndBlock) {
-            uint256 stakingEpoch = stakingContract.stakingEpoch();
+        uint256 stakingEpoch = stakingContract.stakingEpoch();
+        bool earlyEpochEnd = connectivityTracker.isEarlyEpochEnd(stakingEpoch);
```

Comment:
The loading for the connectivityTracker each commited block is a quite a bit overhead,
 we probably can manage to avoid.
Will it be possible to inverse the direction ? so the connectivityTracker tells the BlockReward contract that an early epoch end has to happen ?